test/sim/misc/arceus.js

```diff
@@ -3,38 +3,69 @@
 const common = require('./../../common');
 
+let battle;
 const unimportantPokemon = {species: 'magikarp', moves: ['splash']};
 
 describe(`[Hackmons] Arceus`, function () {
 	it(`in untyped forme should change its type to match the plate held`, function () {
-		const battle = common.gen(4).createBattle([
-			[{species: 'arceus', ability: 'multitype', item: 'flameplate', moves: ['rest']}],
-			[unimportantPokemon],
-		]);
-		assert.deepEqual(battle.p1.active[0].getTypes(), ["Fire"]);
+		battle = common.gen(4).createBattle([[
+			{species: 'arceus', ability: 'multitype', item: 'flameplate', moves: ['rest']},
+		], [
+			unimportantPokemon,
+		]]);
+		const arceus = battle.p1.active[0];
+		assert(arceus.hasType('Fire'));
 	});
 
 	it(`in Steel forme should should be Water-typed to match the held Splash Plate`, function () {
-		const battle = common.gen(4).createBattle([
-			[{species: 'arceussteel', ability: 'multitype', item: 'splashplate', moves: ['rest']}],
-			[unimportantPokemon],
-		]);
-		assert.deepEqual(battle.p1.active[0].getTypes(), ["Water"]);
+		battle = common.gen(4).createBattle([[
+			{species: 'arceussteel', ability: 'multitype', item: 'splashplate', moves: ['rest']},
+		], [
+			unimportantPokemon,
+		]]);
+		const arceus = battle.p1.active[0];
+		assert(arceus.hasType('Water'));
 	});
 
 	it(`in a typed forme should be Normal-typed if no plate is held`, function () {
-		const battle = common.gen(4).createBattle([
-			[{species: 'Arceusfire', ability: 'multitype', item: 'leftovers', moves: ['rest']}],
-			[unimportantPokemon],
-		]);
-		assert.deepEqual(battle.p1.active[0].getTypes(), ["Normal"]);
+		battle = common.gen(4).createBattle([[
+			{species: 'arceusfire', ability: 'multitype', item: 'leftovers', moves: ['rest']},
+		], [
+			unimportantPokemon,
+		]]);
+		const arceus = battle.p1.active[0];
+		assert(arceus.hasType('Normal'));
 	});
 
 	it(`in a typed forme should be Normal-typed despite holding a plate if Arceus does not have the Multitype ability`, function () {
-		const battle = common.gen(4).createBattle([
-			[{species: 'arceusfire', ability: 'truant', item: 'flameplate', moves: ['rest']}],
-			[unimportantPokemon],
-		]);
-		assert.deepEqual(battle.p1.active[0].getTypes(), ["Normal"]);
+		battle = common.gen(4).createBattle([[
+			{species: 'arceusfire', ability: 'truant', item: 'flameplate', moves: ['rest']},
+		], [
+			unimportantPokemon,
+		]]);
+		const arceus = battle.p1.active[0];
+		assert(arceus.hasType('Normal'));
+	});
+
+	it(`should not be able to lose its typing`, function () {
+		battle = common.createBattle([[
+			{species: 'arceus', ability: 'multitype', item: 'flameplate', moves: ['burnup']},
+		], [
+			{species: 'reuniclus', moves: ['soak']},
+		]]);
+		battle.makeChoices();
+		const arceus = battle.p1.active[0];
+		assert(arceus.hasType('Fire'), 'Arceus should not change type.');
+	});
+
+	it.skip(`should use Arceus's real type for Revelation Dance`, function () {
```

Comment:
What happens if Dancer Noivern is Roosting and then copies Revelation Dance?

Reply:
That Noivern uses a Dragon-type Revelation Dance (verified into Color Change). Should a Roosting Tornadus attempt a similar task, its Revelation Dance will be Normal-type.